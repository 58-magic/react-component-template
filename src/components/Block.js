import React, { Component } from 'react';

export default class Block extends Component {
  render() {
    return <div>Block1124</div>;
  }
}
