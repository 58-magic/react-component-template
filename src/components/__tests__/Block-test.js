/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import Block from '../Block';

// TODO
function expectToMatchSnapshot(component) {
  expect(renderer.create(component).toJSON()).toMatchSnapshot();
}

describe('<Block />', () => {
  beforeAll(() => {
    // eslint-disable-next-line no-console
    console.error = error => {
      throw new Error(error);
    };
  });

  it('should have sane defaults', () => {
    expectToMatchSnapshot(<Block />);
  });

  it('should accept className', () => {
    expectToMatchSnapshot(<Block className="block" />);
  });
});
