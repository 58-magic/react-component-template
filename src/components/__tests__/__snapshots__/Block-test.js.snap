// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> should accept className 1`] = `
<div>
  Block
</div>
`;

exports[`<Block /> should have sane defaults 1`] = `
<div>
  Block
</div>
`;
